### PR TITLE
Update GitHub Actions runners to match current (non-beta) up-to-date versions

### LINKED ISSUE #3852: "The macOS 10.15 Actions runner image is being deprecated" -- replace with `macos-11`
See: https://github.blog/changelog/2022-07-20-github-actions-the-macos-10-15-actions-runner-image-is-being-deprecated-and-will-be-removed-by-8-30-22/

> Workflows using the macos-10.15 YAML workflow label should be updated to macos-11, macos-12, or macos-latest

Historically, we've avoided testing against `macos-11` and above because:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/6c5815f4537352c020c41db53a0e07fd094048c4/.github/workflows/tests.yml#L267

However, I'm sure this comment is far out of date by now, so we should make the switch.

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [ ] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [ ] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [ ] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [ ] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

This goes a bit above-and-beyond #3852, but I figured I would update everything together.

As mentioned in f8547fb245d1ed0a05922e8d83b8cc4f4653064a:

> NB: We may also want to test `ubuntu-22.04` and `windows-2022` against both `tests` and `batch-processing`, too, just to be prepared for future deprecations.

I wanted to start by testing `macos-12` in `tests.yml`, but we should maybe add additional commits afterwards for these cases, too.

## Linked issues

Fixes #3852.

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (joshuacwnewton) [inline .github/workflows/tests.yml:267]: `macos-12` passes, so I'll put it back to being part of the scheduled jobs (rather than the PR jobs)

**Review by mguaypaq — APPROVED** [review]
I'm noticing that `.github/workflows/tests.yml` still contains a bunch of references to `ubuntu-18.04`, should some of them also be updated to `ubuntu-20.04`? (Some of them have an explicit comment about wanting to use an older kernel, so probably not those.)

**Comment by joshuacwnewton** [comment]
> I'm noticing that `.github/workflows/tests.yml` still contains a bunch of references to `ubuntu-18.04`, should some of them also be updated to `ubuntu-20.04`? (Some of them have an explicit comment about wanting to use an older kernel, so probably not those.)

Ah, sorry! My rationale for [d6d58e6](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3853/commits/d6d58e622c1ed8facc93465caf0d4cad6213fa2a) wasn't very clear.

For both `create-release.yml` and `test-batch-processing.yml`, I wanted the distributions we test to match the "core" set of distributions we test on PRs in `tests.yml`. I only bumped 18.04 -> 20.04 for `test-batch-processing` because it was mismatched compared to the other 2. 

My thinking is that we would bump the other 18.04-based runners in `tests.yml` only after GitHub Actions no longer offers it? But, maybe there's a different approach we could use?

**Comment by mguaypaq** [comment]
As noted in [the deprecation announcement](https://github.blog/changelog/2022-07-20-github-actions-the-macos-10-15-actions-runner-image-is-being-deprecated-and-will-be-removed-by-8-30-22/), we've entered a 24-hour period where macOS 10.15 runners are unavailable (July 27 from midnight UTC to midnight UTC). This is the missing automated check to merge this PR (after which, we won't be blocked by the automated check for other PRs). I don't have admin privileges for this repo, so I can't bypass it, unfortunately. @joshuacwnewton, would you do the honours?

**Comment by joshuacwnewton** [comment]
> This is the missing automated check to merge this PR (after which, we won't be blocked by the automated check for other PRs)

I think this might be because `macos-10.15` has been removed from the CI workflow entirely? 

This kind of "Waiting for status to be reported" message tends to happen when a status check is marked as "required" in the branch protection settings, but isn't run. 

image

If I remove this requirement (since it's going to be replaced with `macos-11` anyway), this PR should become mergeable without admin privileges. (Done!)

**Comment by mguaypaq** [comment]
And now that the PR is merged, I added macOS 11 as a required check:
image

**Pull request merged**
